Data-Structure/Algorithm-Design-Part-1/DictionaryUsingBST/BSTNode.java

```diff
@@ -0,0 +1,55 @@
+package DictionaryUsingBST;
+
+/**
+ * @author Niraj Gupta
+ * Class to create a node which is inserted in the Tree. Each node has key and value and reference to their 
+ * left and right child node.
+ * 
+ * @param <K> : The Data Type assigned to the key of the node.

```

Comment:
class level variables can not be categorized as params. 

And since they are explained in constructor specifications we could have avoided adding details here.

Reply:
The class is generic and the parameters are passed to the class so I think it should be mentioned above the class. And i have seen it in an example.